### ISSUE TITLE
Clippy fails to apply fixes

### ISSUE THREAD

**Issue by dee-see** [rust-lang/rust#90184]
I'm posting this here as suggested by clippy. Interestingly, running `cargo clippy` without the `--fix` outputs nothing (suggesting there's nothing to fix)

Also if that's of any interest, the code clippy is trying to fix is inside a `#[tokio::test]`.

```shell
$ cargo clippy --fix
warning: failed to automatically apply fixes suggested by rustc to crate `dns_fuzzer`

after fixes were automatically applied the compiler reported errors within these files:

  * src/generator.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0308]: mismatched types
   --> src/generator.rs:617:21
    |
617 |                 (0..number_of_tokens).map(|i| char::from(start_char + i).to_string()),
    |                     ^^^^^^^^^^^^^^^^ expected integer, found `&u8`
    |
help: consider dereferencing the borrow
    |
617 |                 (0..*number_of_tokens).map(|i| char::from(start_char + i).to_string()),
    |                     +

error[E0277]: the trait bound `usize: std::convert::From<&u8>` is not satisfied
   --> src/generator.rs:620:36
    |
620 |             let number_of_tokens = usize::from(number_of_tokens);
    |                                    ^^^^^^^^^^^ the trait `std::convert::From<&u8>` is not implemented for `usize`
    |
    = help: the following implementations were found:
              <usize as std::convert::From<bool>>
              <usize as std::convert::From<mio::token::Token>>
              <usize as std::convert::From<std::num::NonZeroUsize>>
              <usize as std::convert::From<tokio::runtime::thread_pool::idle::State>>
            and 2 others
note: required by `std::convert::From::from`
   --> /home/dee-see/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/convert/mod.rs:371:5
    |
371 |     fn from(_: T) -> Self;
    |     ^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
Original diagnostics will follow.

warning: useless use of `vec!`
   --> src/generator.rs:613:33
    |
613 |         for number_of_tokens in vec![5u8, 10u8] {
    |                                 ^^^^^^^^^^^^^^^ help: you can use a slice directly: `&[5u8, 10u8]`
    |
    = note: `#[warn(clippy::useless_vec)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_vec

warning: `dns-fuzzer` (lib test) generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 1.14s
```

### Meta

`rustc --version --verbose`:
```
rustc 1.56.0 (09c42c458 2021-10-18)
binary: rustc
commit-hash: 09c42c45858d5f3aedfa670698275303a3d19afa
commit-date: 2021-10-18
host: x86_64-unknown-linux-gnu
release: 1.56.0
LLVM version: 13.0.0
```

**Comment by Noratrieb** [comment]
An old issue without a reproduction, closing.
